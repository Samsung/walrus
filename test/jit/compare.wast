(module
(func (export "test1") (result i32 i32 i32 i32) (local i64)
  i64.const 0x0
  local.set 0

  i64.const 0xff00000000
  i64.eqz

  i64.const 0x1
  i64.eqz

  local.get 0
  i64.eqz

  local.get 0
  i64.eqz
  if (result i32)
     block (result i32)
       i32.const 3

       local.get 0
       i64.eqz
       br_if 0

       drop
       i32.const 4
    end
  else
    i32.const 6
  end

  (; 0, 0, 1, 3 ;)
)
)

(assert_return (invoke "test1") (i32.const 0) (i32.const 0) (i32.const 1) (i32.const 3))
