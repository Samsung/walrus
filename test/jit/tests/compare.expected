[[[[[[[  Function   0  ]]]]]]]
Local 0: mapped to: 0 range [1 - 12]
Number of locals reduced from 1 to 1
0: Opcode: i64.const
  Result[0]: i64
1: Opcode: local.set
  Index: 0
  Param[0]: ref:0(0)
2: Opcode: i64.const
  Result[0]: i64
3: Opcode: i64.eqz
  Param[0]: ref:2(0)
  Result[0]: i32
4: Opcode: i64.const
  Result[0]: i64
5: Opcode: i64.eqz
  Param[0]: ref:4(0)
  Result[0]: i32
6: Opcode: local.get
  Index: 0
  Result[0]: i64
7: Opcode: i64.eqz
  Param[0]: ref:6(0)
  Result[0]: i32
8: Opcode: local.get
  Index: 0
  Result[0]: i64
9: Opcode: i64.eqz
  Param[0]: ref:8(0)
  Result[0]: i32
10: Opcode: br_unless
  Jump to: 19
  Param[0]: ref:9(0)
11: Opcode: i32.const
  Result[0]: i32
12: Opcode: local.get
  Index: 0
  Result[0]: i64
13: Opcode: i64.eqz
  Param[0]: ref:12(0)
  Result[0]: i32
14: Opcode: br_if
  Jump to: 17
  Param[0]: ref:13(0)
15: Opcode: drop
  Param[0]: ref:11(0)
16: Opcode: i32.const
  Result[0]: i32
17: Label: result_count: 1 preserved_count: 3
  Jump from: 14
18: Opcode: br
  Jump to: 21
19: Label: result_count: 0 preserved_count: 3
  Jump from: 10
20: Opcode: i32.const
  Result[0]: i32
21: Label: result_count: 1 preserved_count: 3
  Jump from: 18
  Param[0]
  Param[1]
  Param[2]
  Param[3]
    20(0) instruction
    11(0) instruction
    16(0) instruction
22: Opcode: return
  Param[0]: ref:3(0)
  Param[1]: ref:5(0)
  Param[2]: ref:7(0)
  Param[3]: ref:21(3)
------------------------------
Frame size: 0, Arguments size: 32 (total: 32)
0: Opcode: i64.const
  Result[0]: i64 unused
1: Opcode: local.set
  Addr: sp[16]
  Param[0]: i64 const:0
2: Opcode: i64.const
  Result[0]: i64 unused
3: Opcode: i64.eqz
  Param[0]: i64 const:2
  Result[0]: i32 sp[0]
4: Opcode: i64.const
  Result[0]: i64 unused
5: Opcode: i64.eqz
  Param[0]: i64 const:4
  Result[0]: i32 sp[4]
6: Opcode: local.get
  Addr: sp[16]
  Result[0]: i64 unused
7: Opcode: i64.eqz
  Param[0]: i64 sp[16]
  Result[0]: i32 sp[8]
8: Opcode: local.get
  Addr: sp[16]
  Result[0]: i64 unused
9: Opcode: i64.eqz
  Param[0]: i64 sp[16]
  Result[0]: i32 unused
10: Opcode: br_unless
  Jump to: 19
  Param[0]: i32 unused
11: Opcode: i32.const
  Result[0]: i32 sp[12]
12: Opcode: local.get
  Addr: sp[16]
  Result[0]: i64 unused
13: Opcode: i64.eqz
  Param[0]: i64 sp[16]
  Result[0]: i32 unused
14: Opcode: br_if
  Jump to: 17
  Param[0]: i32 unused
15: Opcode: drop
  Param[0]: i32 sp[12]
16: Opcode: i32.const
  Result[0]: i32 sp[12]
17: Label: result_count: 1 preserved_count: 3
  Jump from: 14
18: Opcode: br
  Jump to: 21
19: Label: result_count: 0 preserved_count: 3
  Jump from: 10
20: Opcode: i32.const
  Result[0]: i32 sp[12]
21: Label: result_count: 1 preserved_count: 3
  Jump from: 18
  Param[0]
  Param[1]
  Param[2]
  Param[3]
    20(0) instruction
    11(0) instruction
    16(0) instruction
22: Opcode: return
  Param[0]: i32 sp[0]
  Param[1]: i32 sp[4]
  Param[2]: i32 sp[8]
  Param[3]: i32 sp[12]

test1() => i32:0, i32:0, i32:1, i32:3
